### DISCUSSION TITLE
How to implement MPI_ALLREDUCE in AMREX?

### DISCUSSION THREAD

**Discussion by solarmaterial** [AMReX-Codes/amrex#4004] [Q&A] [answered]
Hey AMReX team,
  When dealing with parallel computation, each processor computes a variable, e.g., the time step `dt_Local`. However, computaion on all the processors should advance with the same time step, therefore, we need the minimum `dt_Local ` on all the processors. This can be realized with the MPI_ALLREDUCE, but how to implement in AMREX? 
 
  Looking forward to your reply!

Yours sincerely,
solar

**Comment by atmyers** [comment, marked as answer]
Hi solar,

`AMReX_ParallelDescriptor.H` contains a number of wrappers around MPI functions like that, including for example [ReduceRealMin](https://amrex-codes.github.io/amrex/doxygen/namespaceamrex_1_1ParallelDescriptor.html#a95eed49d43cbcd3d04dd6afcfc1c5935), which can be used to do what you want here.   

-Andrew

**Reply by solarmaterial** [reply]
Thank you for your replies! I'll try it.
